### ISSUE TITLE
[Feature Request] Adjustabe audio bitrates depending on channels / DynamicNormalization

### ISSUE THREAD

**Issue by MBCD2000** [RandomEngy/VidCoder#714]
I would like to configure/preset which bitrate is used, by amount of channels.
There is a "auto"-function, but in my case it used 1536kbps for a 5.1 channel audio. This is way too high, as 640 kbps are enough,
It would be nice to have a possibility to define a mapping depending on the amount of channels => 6ch = 640kbps, 2 ch = 224kbps, ...
It should always be selected by the existing stream channels and not what was set in UI (channel-downmix).

If I select a conversation to 5.1, a 7.1 source should be downmixed to 640kbps, but a 2 channel source should be left as 2 channel, but with only 224kbps.

Right now it does (at bitrate set to 192 at 5.1):
DTS 2.0 (768kbps) ->AC3  Dolby Surround 192kbps
DTS-MA 5.1 -> AC3  5.1 224 kbps

So bitrate is fine for 2.0 channel, but way too low for 5.1 channel.

--------------------------------------------------------------------------

Also it would be fine to get the DynamicNormalization-Filter to reduce dynamic-range and adjusting all audio to an identical (dialog) level.
Because watching movies at night is sometimes difficult and those implemented "nightmodes" (some players have) do not work proper.

I want to reduce dynamic-range to e.g. 6 db and keep dialoglevel (or at least overall-level) at -9db ... I hope you get my explanation.

**Comment by RandomEngy** [comment]
Yes, Dynamic Normalization would be something to ask the HandBrake team about.

And the auto bitrate is something that HandBrake controls. I could see offering options to manually tweak this, but it would be a complicated feature. Different codecs need different bitrates, and not all channels are equal. I think LFE channels typically need less data to encode. It would be a burden to maintain as codecs are added. I think recommending changes to the HandBrake team for the default bitrates would probably be a better bet, or using manual settings.

**Comment by MBCD2000** [comment]
Thank you for answering, very nice you take your time for it!
Yes, if you do autobitrate correctly, you should to do per codec, if you claim to do it "right".
Basically you are right, the are plenty of speaker-configurations possible, but mostly you still have common usage of them, so 6ch is mostly 5.1.

Or what I would say, you still have to define between mono, stereo and multichannel, this would be some basics. Maybe this is some easy option to implement to make automatic tasks a little bit more precise.

**Comment by RandomEngy** [comment]
Well it's not just as simple as channel count. LFE channels have different bitrates sometimes and I don't know what other oddities are lurking in the different channel layouts. Also there would be the issue of having the default suggested values for per-channel bitrate match what's automatically given by the HandBrake API for a default bitrate. And the fact that an entered per-channel bitrate might not add up to a supported overall bitrate, which is what we have to give to the encoder. And the fact that when the user enters the values for the LFE and main channels, that's not actually how much data will be given to them.

I would consider implementing it if the HandBrake API exposed it.
